### PR TITLE
API-14989 Added reasonCode to DeviceRequest datamart and transformer.

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**
https://vajira.max.gov/browse/API-14989

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java [hunk 1]
```diff
@@ -32,13 +38,33 @@ Meta meta(Instant dmLastUpdated) {
...
+                .code(reasonCode.code())
+                .display(reasonCode.display())
+                .build()
+                .asList())
+        .text(reasonCode.display())
+        .build();
+  }
+
+  List<CodeableConcept> reasonCodes(List<DatamartDeviceRequest.ReasonCode> reasonCodes) {
+    return emptyToNull(Safe.stream(reasonCodes).map(rc -> reasonCode(rc)).collect(toList()));
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java:60]: In one of the more recent java versions, you can replace `collect(toList())` with just `toList()`

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java [hunk 2]
```diff
@@ -32,13 +38,33 @@ Meta meta(Instant dmLastUpdated) {
...
 
+  CodeableConcept reasonCode(DatamartDeviceRequest.ReasonCode reasonCode) {
+    if (reasonCode == null
+        || allBlank(reasonCode.code(), reasonCode.codeVersion(), reasonCode.display())) {
+      return null;
+    }
+    return CodeableConcept.builder()
+        .coding(
+            Coding.builder()
+                .system("http://www.va.gov/Terminology/VistADefinedTerms/9000010-.07")
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java:49]: @nmoisan-va can you confirm this is the right system

Comment (nmoisan-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java:48]: 🤔 Is that what we have in Encounter and other places that use ICD codes?

Comment (nmoisan-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java:48]: I think it should be more similar to [Condition ](https://github.com/department-of-veterans-affairs/health-apis-data-query/blob/master/data-query/src/main/java/gov/va/api/health/dataquery/service/controller/condition/R4ConditionTransformer.java#L133).

with the exception that will not ever provide a SNOMED code

Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java:48]: It's the system currently used for Encounter `reasonCode`

Comment (taylorrassmannlibertyits) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/devicerequest/R4DeviceRequestTransformer.java]: Updated to mirror Condition

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**